### ISSUE TITLE
Losing Trace logging on closing application

### ISSUE THREAD

**Issue by Inlustris** [sgryphon/essential-diagnostics#37]
Hello i'm happy to work with your tracelistening tooling. I only walk in to a problem when working with our old console applications. 

The console applications closes before sending the async logging. By updating the batchsize setting to 0 i can manage to send the logging in time before the application turns off. The problem with this is that the application only runs when the Seq log server is available. 

If the server is offline with batchsize of 0 the application will crash on every trace happening. 

Is it possible to set the batchsize to 0 with errorhandling? 
OR 
Create a possibility to wait for a full flush of the SeqTraceListener?

**Comment by Inlustris** [comment]
I allready made a change for it. Can you give me push permissions so i can create a branch and start a pull request?

**Comment by sgryphon** [comment]
Thanks for the pull request. I added comments.

As a work around, you can use batchsize = 1, which will send each message separately, although it will still be async so you could still lose the most recent messages.

Implementing Flush() would also be a good solution. I.e. override Flush() in the listener so that it sends the current batch immediately (and waits for completion).  This is probably the correct solution.   In your console app you can then call System.Diagnotics.Trace.Flush(); before exiting (which will then flush all pending messages).  

That would allow both the performance benefits of batching, the retries (if connection issues are only temporary), and to log everything before exiting.

The implementation of Flush() can be a separate pull request.

**Comment by Inlustris** [comment]
I have tried the flush solution with a batchsize of 1. This sadly is not working because of the async processing of the flush. The only current work around is implementing a task.delay at the end of the proces.

To have it work indeed i think the flush() should be implemented as an override which sync waits on completion before exiting the application. 

With the recent pullrequest i can fix application crashes when the Seq server is not available and secondary get a solution for this too by sending it sync.
